pkg/gitops/profile.go

```diff
@@ -0,0 +1,138 @@
+package gitops
+
+import (
+	"context"
+	"github.com/kris-nova/logger"
+	"github.com/pkg/errors"
+	"github.com/spf13/afero"
+	"github.com/weaveworks/eksctl/pkg/git"
+	"github.com/weaveworks/eksctl/pkg/gitops/fileprocessor"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+const (
+	cloneDirPrefix = "quickstart-"
+)
+
+// Profile represents a GitOps profile
+type Profile struct {
+	Processor fileprocessor.FileProcessor
+	Path      string
+	GitOpts   GitOptions
+	GitCloner git.Cloner
+	FS        afero.Fs
+	IO        afero.Afero
+}
+
+// GitOptions holds options for cloning a git repository
+type GitOptions struct {
+	URL    string
+	Branch string
+}
+
+// Generate clones the specified Git repo in a base directory and generates overlays if the Git repo
+// points to a profile repo
+func (p *Profile) Generate(ctx context.Context) error {
+	logger.Info("cloning repository %q:%s", p.GitOpts.URL, p.GitOpts.Branch)
+	clonedDir, err := p.GitCloner.CloneRepo(cloneDirPrefix, p.GitOpts.Branch, p.GitOpts.URL)
```

Comment:
Where is `clonedDir` deleted? I can't find it.

Reply:
(Maybe the cloner should output a cleanup function to delete it?)

Reply:
Good point. Done.

Reply:
Did you?

Reply:
Ah, you meant you deleted it later, not that you added the cleanup function